source/slang/core.meta.slang

```diff
@@ -1001,8 +1001,12 @@ for (int tt = 0; tt < kBaseTextureTypeCount; ++tt)
             static const struct {
                 char const* genericPrefix;
                 char const* elementType;
+                char const* outputType;
             } kGatherExtensionCases[] = {
-                { "__generic<T, let N : int>", "vector<T,N>" },
+                { "__generic<T, let N : int>", "vector<T,N>", "vector<T, 4>" },
+                { "", "float", "vector<float, 4>" },
+                { "", "int" , "vector<int, 4>"},
+                { "", "uint", "vector<uint, 4>"},
```

Comment:
We probably need/want to handle all the scalar types that are valid for `Texture2D<T>` here, but these three will cover 99.9% of cases so I'm fine with it for a near-term fix.